### PR TITLE
[EAM-102] fix parent asset/position not being saved as dependent

### PR DESCRIPTION
**Pull request by booua**

### CODE REVIEW

File: src/ui/pages/equipment/asset/AssetHierarchy.js [hunk 1]
```diff
@@ -28,43 +86,84 @@ class AssetHierarchy extends Component {
...
+                        </div>
+                    </div>
 
                     <EAMBarcodeInput updateProperty={value => updateEquipmentProperty('hierarchyPrimarySystemCode', value)} right={0} top={20}>
                         <EAMAutocomplete
                             children = {children}
                             elementInfo={assetLayout.fields['primarysystem']}
                             value={equipment.hierarchyPrimarySystemCode}
-                            updateProperty={updateEquipmentProperty}
+                            updateProperty={(key, value) => this.updateDependencyProperty(key,value)}
```
Comment (ricardo-j-almeida) [inline src/ui/pages/equipment/asset/AssetHierarchy.js]: Forgot equipment

File: src/ui/pages/equipment/position/PositionHierarchy.js [hunk 2]
```diff
@@ -3,9 +3,67 @@ import EISPanel from 'eam-components/dist/ui/components/panel';
...
+        }
+    }
+
+    updateDependencyProperty = (key, value, equipment) => {
+        let { updateEquipmentProperty } = this.props
+        //inversed map of hierarchy properties. Asset Dependency has the highest priority.
+        
+        let hierarchyMap = {
+            "hierarchyPrimarySystemCode": "hierarchyPrimarySystemDependent",
+            "hierarchyPositionCode": "hierarchyPositionDependent",
```
Comment (ricardo-j-almeida) [inline src/ui/pages/equipment/position/PositionHierarchy.js]: Asset?

File: src/ui/pages/equipment/position/PositionHierarchy.js [hunk 3]
```diff
@@ -3,9 +3,67 @@ import EISPanel from 'eam-components/dist/ui/components/panel';
...
+        if (Object.keys(hierarchyMap).includes(key)) {
+            if (value && equipment) {
+                //index of currently checked checkbox - if no value is present indexExistant = -1
+                let indexExistant = Object.values(hierarchyMap).map(prop => equipment[prop])
+                    .findIndex(value => value && value !== false.toString())
+                //index of edited hierarchy field
+                let indexAfter = Object.keys(hierarchyMap).findIndex(prop => prop === key)
+
+                if (indexAfter > indexExistant) {
+                    updateEquipmentProperty(hierarchyMap[key], value ? true.toString(): false.toString())
```
Comment (ricardo-j-almeida) [inline src/ui/pages/equipment/position/PositionHierarchy.js]: Where do we unset the existent one?

File: src/ui/pages/equipment/position/PositionHierarchy.js (hunk 2 above)
Comment (booua) [reply src/ui/pages/equipment/position/PositionHierarchy.js]: there is no asset dependency in position

File: src/ui/pages/equipment/asset/AssetHierarchy.js (hunk 1 above)
Comment (booua) [reply src/ui/pages/equipment/asset/AssetHierarchy.js]: indeed

File: src/ui/pages/equipment/position/PositionHierarchy.js (hunk 3 above)
Comment (booua) [reply src/ui/pages/equipment/position/PositionHierarchy.js]: should be fine now

**Pull request merged**